### PR TITLE
Blog app - creating a data model

### PR DESCRIPTION
**Pull request by lincoln1883**
For this project requirement i completed the following task:
- [x] Build your project schema.
- [x] Create and run the necessary migration files.
- [x] Table and column names should match the ERD diagram.
  - [x] note: photo for users table should be a link to an image
- [x] Foreign keys should be included.
  - [x] Pay attention that in posts table the column holding the foreign key to the users table should be named author_id
- [x] All foreign key columns should have a corresponding index.

### CODE REVIEW

File: db/migrate/20230815012535_create_posts.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class CreatePosts < ActiveRecord::Migration[7.0]
+  def change
+    create_table :posts do |t|
+      t.string :title
+      t.text :text
+      t.integer :commentsCounter
+      t.integer :likesCounter
+      t.references :user, null: false, foreign_key: true
```
Comment (Olamarx) [inline db/migrate/20230815012535_create_posts.rb:8]: - [ ] This refresh should use the user key and where user is, it is supposed to be author, something like `t.references :author, foreign_key: { to_table: 'users' }`

File: db/migrate/20230815012535_create_posts.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class CreatePosts < ActiveRecord::Migration[7.0]
+  def change
+    create_table :posts do |t|
+      t.string :title
+      t.text :text
+      t.integer :commentsCounter
+      t.integer :likesCounter
```
Comment (Olamarx) [inline db/migrate/20230815012535_create_posts.rb:7]: - [ ] I will encourage you to use snake_case `likes_counter` primarily for consistency, readability, and convention. This choice of naming convention is part of Rails' philosophy of "Convention over Configuration," which aims to provide sensible defaults and reduce the amount of configuration required in development.

File: db/migrate/20230815013913_create_comments.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class CreateComments < ActiveRecord::Migration[7.0]
+  def change
+    create_table :comments do |t|
+      t.text :text
+      t.references :user, null: false, foreign_key: true
```
Comment (Olamarx) [inline db/migrate/20230815013913_create_comments.rb:5]: - [ ] I will encourage you to use author here to and refrence the user `t.references :author, foreign_key: { to_table: 'users' }`

File: db/migrate/20230815013139_create_likes.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class CreateLikes < ActiveRecord::Migration[7.0]
+  def change
+    create_table :likes do |t|
+      t.references :post, null: false, foreign_key: true
+      t.references :user, null: false, foreign_key: true
```
Comment (Olamarx) [inline db/migrate/20230815013139_create_likes.rb:5]: - [ ] Here too for the author

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @lincoln1883, 

Good job so far, indeed you have spent time on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- Linters passed ✔️
- All migration generated ✔️
- Readme file professional ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20230815013913_create_comments.rb (hunk 3 above)
Comment (lincoln1883) [reply db/migrate/20230815013913_create_comments.rb:5]: Thank you however this was not written in the project requirements.
Screenshot from 2023-08-15 00-22-07
Screenshot from 2023-08-15 00-17-09

File: db/migrate/20230815013139_create_likes.rb (hunk 4 above)
Comment (lincoln1883) [reply db/migrate/20230815013139_create_likes.rb:5]: Thank you however this was not written in the project requirements.
Screenshot from 2023-08-15 00-22-07
Screenshot from 2023-08-15 00-17-09

File: db/migrate/20230815012535_create_posts.rb (hunk 1 above)
Comment (lincoln1883) [reply db/migrate/20230815012535_create_posts.rb:8]: I realised my mistake and made changes in a later commit.
[db/migrate/20230815015726_change_user_id_to_author_id.rb](https://github.com/lincoln1883/blog-app/pull/1/files/da1c5b74e74d007732c36179856bdb4652643b20#diff-32ee16f03eb11201f5b2df14a956bbbf64227dc413edbed8f48275b08a01a7da)

File: db/migrate/20230815013913_create_comments.rb (hunk 3 above)
Comment (Olamarx) [reply db/migrate/20230815013913_create_comments.rb:5]: - [ ] Yes, it is not written but traditionally, javascript use camel_case and RoR uses snake_case, it is a convention, I will approve your project now but I will encourage you to adopt the professional approach.

Comment (Olamarx) [reply db/migrate/20230815013913_create_comments.rb:5]: - [ ] It is there, you can see that authorId is referencing the user.

File: db/migrate/20230815013139_create_likes.rb (hunk 4 above)
Comment (Olamarx) [reply db/migrate/20230815013139_create_likes.rb:5]: > Thank you however this was not written in the project requirements. Screenshot from 2023-08-15 00-22-07 Screenshot from 2023-08-15 00-17-09

- [ ] It is there, you can see that authorId is referencing the user. Kindly fix it and others too.

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @lincoln1883, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- Linters passed ✔️
- All migration generated ✔️
- Readme file professional ✔️

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by brainconnect93 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @lincoln1883, 👋 👋  

Your project is complete! Great job for following the project requirement🥇and implementing the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 




## Good Points 👍

- Descriptive Pull Request title & summary. ✔️
- Created migration file for `users`, `posts`, `comments` & `likes` ✔️
- Migrate all file to the schema ✔️ 
- Good use of Git flow ✔️

## Optional changes

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brainconnect93 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**